lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,44 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.respond_to?(:cause) && exception.cause
```

Comment:
If you use a real exception, please do not stub anything on it.  Instead, you can set the exception by passing it as an argument to `Exception.new`.  You can set the backtrace by calling `exception.set_backtrace`.  You can set the cause by simulating the conditions where ruby sets the cause:

```
irb(main):023:0> ex = begin
irb(main):024:1*   begin
irb(main):025:2*     raise "root cause"
irb(main):026:2>   rescue
irb(main):027:2>     raise "from rescue"
irb(main):028:2>   end
irb(main):029:1> rescue => e
irb(main):030:1>   e
irb(main):031:1> end
=> #<RuntimeError: from rescue>
irb(main):032:0> ex.cause
=> #<RuntimeError: root cause>
```